metricbeat/helper/server/http/http.go

```diff
@@ -66,8 +72,9 @@ func NewHttpServer(mb mb.BaseMetricSet) (server.Server, error) {
 	}
 
 	httpServer := &http.Server{
-		Addr:    fmt.Sprintf("%s:%d", config.Host, config.Port),
-		Handler: http.HandlerFunc(h.handleFunc),
+		Addr:      fmt.Sprintf("%s:%d", config.Host, config.Port),
+		Handler:   http.HandlerFunc(h.handleFunc),
+		TLSConfig: tlsConfig.BuildModuleConfig(config.Host),
```

Comment:
I would set this conditionally based on whether `tlsConfig` is `nil` or not. The ensures that it respects the `config.TLS.IsEnabled` value because `LoadTLSServerConfig` will return nil when it's not enabled.

Reply:
Agreed

Reply:
I meant adding a check like

```
if tlsConfig != nil {
    httpServer.TLSConfig = tlsConfig.BuildModuleConfig(config.Host)
}
```

This ensure that the http server's `TLConfig` is non-nil when the user enabled it in their config.